### PR TITLE
samples: buck: rework BUCK example into 4 separate ones

### PR DESCRIPTION
**Pull request by jaz1-nordic**

### CODE REVIEW

File: samples/buck/enable_bucks_using_pin/src/main.c [hunk 1]
```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2022-2023 Nordic Semiconductor ASA
```
Comment (wilu-nordic) [inline samples/buck/enable_bucks_using_pin/src/main.c:2]:
```suggestion
 * Copyright (c) 2023 Nordic Semiconductor ASA
```

Comment (wilu-nordic) [reply samples/buck/enable_bucks_using_pin/src/main.c:2]: Please run pre-commit for all affected files. A common mistake is the year in the license header.

File: samples/buck/output_voltage_advanced/README.rst [hunk 2]
```diff
@@ -0,0 +1,102 @@
+.. _buck_sample:
```
Comment (wilu-nordic) [inline samples/buck/output_voltage_advanced/README.rst:1]:
```suggestion
.. _buck_output_voltage_advanced_sample:
```
The same for other samples.

File: samples/buck/enable_bucks_using_pin/src/main.c [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+#include <npmx_buck.h>
+#include <npmx_driver.h>
+#include <npmx_gpio.h>
+
+#define LOG_MODULE_NAME pmic_buck
```
Comment (wilu-nordic) [inline samples/buck/enable_bucks_using_pin/src/main.c:13]:
```suggestion
#define LOG_MODULE_NAME pmic_buck_output_voltage_advanced
```

File: samples/buck/output_voltage_advanced/README.rst [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+BUCK - set advanced output voltage
+####
+
+.. contents::
+   :local:
+   :depth: 2
+
+The BUCK samples demonstrates how to control PMIC BUCKs using npmx drivers.
```
Comment (wilu-nordic) [inline samples/buck/output_voltage_advanced/README.rst:10]:
```suggestion
The BUCK Output Voltage Advanced sample demonstrates how to control PMIC BUCKs using npmx drivers.
```

@annakielar should it be here with uppercase?

File: samples/buck/output_voltage_advanced/README.rst [hunk 5]
```diff
@@ -0,0 +1,102 @@
+.. _buck_sample:
+
+BUCK - set advanced output voltage
+####
```
Comment (wilu-nordic) [inline samples/buck/output_voltage_advanced/README.rst:4]:
```suggestion
BUCK - set advanced output voltage
##################################
```

File: samples/buck/output_voltage_advanced/README.rst [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+       - P0.27
+     * - VOUT2 & GND
+       - External supply (P21)
+       - External supply (P21)
+       - External supply (P21)
+
+#. Make the following connections on the nPM1300 EK:
+
+   * Connect a USB power supply to the **J3** connector.
+   * Connect a suitable battery to the **J2** connector.
```
Comment (wilu-nordic) [inline samples/buck/output_voltage_advanced/README.rst:64]: Battery is not required for this sample

**Comment by wilu-nordic** [comment]
After sphinx generation, there is no BUCK sample

image


One solution is to not create buck subdirectory, the other solution is to fix the sphinx build system to include subdirectories.
Now only samples from the `samples` subdirectory are used:
```
src_dir = NPMX_ZEPHYR_BASE / "samples"
dst_dir = NPMX_ZEPHYR_BASE / "doc" / "samples"
```

**Comment by masz-nordic** [comment]
I think shorter names for samples, like `buck_pins`, `buck_output_simple`, `buck_output_advanced` and `buck_retention` would be better.
@wilu-nordic @annakielar what do you think?

**Comment by wilu-nordic** [comment]
> I think shorter names for samples, like `buck_pins`, `buck_output_simple`, `buck_output_advanced` and `buck_retention` would be better. @wilu-nordic @annakielar what do you think?

Shorter names are fine for me.

File: samples/buck_output_simple/README.rst [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+The sample supports the following development kits:
+
+.. table-from-sample-yaml::
+
+The sample also requires an nPM1300 EK.
+
+Overview
+********
+
+This sample uses the npmx BUCK driver to simple set output voltage for the buck converter.
```
Comment (wilu-nordic) [inline samples/buck_output_simple/README.rst:24]:
```suggestion
This sample uses the npmx BUCK driver to set the output voltage for the buck converter.
```

File: samples/buck_output_advanced/README.rst [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+
+Configuration
+*************
+
+|config|
+
+Building and running
+********************
+
+.. |sample path| replace:: :file:`samples/buck`
```
Comment (wilu-nordic) [inline samples/buck_output_advanced/README.rst:82]:
```suggestion
.. |sample path| replace:: :file:`samples/buck_output_advanced`
```

File: samples/buck_pins/README.rst [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+
+The sample also requires an nPM1300 EK.
+
+Overview
+********
+
+This sample uses the npmx BUCK driver to control enable mode for the buck converter with GPIO.
+GPIO3 (:c:enumerator:`NPMX_GPIO_INSTANCE_3`) is set to the input mode and is configured to be used as an enable mode selector for BUCK1 (:c:enumerator:`NPMX_BUCK_INSTANCE_1`).
+The output voltage of BUCK1 is set to 3.3 V (:c:enumerator:`NPMX_BUCK_VOLTAGE_3V3`).
+When you change the input state of the selected GPIO, the voltage in the BUCK1 output will change from 0 V to the previously selected voltage.
```
Comment (wilu-nordic) [inline samples/buck_pins/README.rst:27]:
```suggestion
This sample uses the npmx BUCK driver to control the enable mode for the buck converter with GPIO.
GPIO3 (:c:enumerator:`NPMX_GPIO_INSTANCE_3`) is set to the input mode and is configured to be used as an enable mode selector for BUCK1 (:c:enumerator:`NPMX_BUCK_INSTANCE_1`).
The output voltage of BUCK1 is set to 3.3 V (:c:enumerator:`NPMX_BUCK_VOLTAGE_3V3`).
When you change the input state of the selected GPIO, the voltage in the BUCK1 output will change from 0 V to the previously selected voltage.
```

File: samples/buck_pins/README.rst [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+
+Configuration
+*************
+
+|config|
+
+Building and running
+********************
+
+.. |sample path| replace:: :file:`samples/buck`
```
Comment (wilu-nordic) [inline samples/buck_pins/README.rst:83]:
```suggestion
.. |sample path| replace:: :file:`samples/buck_pins`
```

File: samples/buck_pins/boards/nrf5340dk_nrf5340_cpuapp.overlay [hunk 11]
```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2022-2023 Nordic Semiconductor ASA
```
Comment (wilu-nordic) [inline samples/buck_pins/boards/nrf5340dk_nrf5340_cpuapp.overlay:2]:
```suggestion
 * Copyright (c) 2023 Nordic Semiconductor ASA
```

All new samples should be from the 2023 year.

File: samples/buck_retention/README.rst [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+
+The sample also requires an nPM1300 EK.
+
+Overview
+********
+
+This sample uses the npmx BUCK driver to set retention voltage for the buck converter with GPIO.
+GPIO1 (:c:enumerator:`NPMX_GPIO_INSTANCE_1`) is set to the input mode and is configured to be used as a retention input selector for BUCK1 (:c:enumerator:`NPMX_BUCK_INSTANCE_1`).
+The normal mode voltage is set to 1.7 V (:c:enumerator:`NPMX_BUCK_VOLTAGE_1V7`) and the retention voltage to 3.3 V (:c:enumerator:`NPMX_BUCK_VOLTAGE_3V3`).
+When you change the input state of the selected GPIO, the voltage in the BUCK1 output will change.
```
Comment (wilu-nordic) [inline samples/buck_retention/README.rst:27]:
```suggestion
This sample uses the npmx BUCK driver to set the retention voltage for the buck converter with GPIO.
GPIO1 (:c:enumerator:`NPMX_GPIO_INSTANCE_1`) is set to the input mode and is configured to be used as a retention input selector for BUCK1 (:c:enumerator:`NPMX_BUCK_INSTANCE_1`).
The normal mode voltage is set to 1.7 V (:c:enumerator:`NPMX_BUCK_VOLTAGE_1V7`) and the retention voltage to 3.3 V (:c:enumerator:`NPMX_BUCK_VOLTAGE_3V3`).
When you change the input state of the selected GPIO, the voltage in the BUCK1 output will change.
```

File: samples/buck_retention/README.rst [hunk 13]
```diff
@@ -0,0 +1,104 @@
...
+
+Configuration
+*************
+
+|config|
+
+Building and running
+********************
+
+.. |sample path| replace:: :file:`samples/buck`
```
Comment (wilu-nordic) [inline samples/buck_retention/README.rst:83]:
```suggestion
.. |sample path| replace:: :file:`samples/buck_retention`
```

File: samples/buck_output_advanced/sample.yaml [hunk 14]
```diff
@@ -0,0 +1,31 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+sample:
+  description: Sample for nPM1300 PMIC BUCKs
```
Comment (wilu-nordic) [inline samples/buck_output_advanced/sample.yaml:8]:
```suggestion
  description: Buck output advanced sample
```

File: samples/buck_output_simple/sample.yaml [hunk 15]
```diff
@@ -0,0 +1,30 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+sample:
+  description: Sample for nPM1300 PMIC BUCKs
```
Comment (wilu-nordic) [inline samples/buck_output_simple/sample.yaml:8]:
```suggestion
  description: Buck output simple sample
```

File: samples/buck_output_simple/sample.yaml [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+common:
+  integration_platforms:
+    - nrf5340dk_nrf5340_cpuapp
+    - nrf52840dk_nrf52840
+  platform_allow: nrf5340dk_nrf5340_cpuapp nrf52840dk_nrf52840 nrf52dk_nrf52832
+  tags: pmic
+  timeout: 30
+
+tests:
+  sample.buck.set_buck_voltage_eng_c:
```
Comment (wilu-nordic) [inline samples/buck_output_simple/sample.yaml:20]:
```suggestion
  sample.buck_output_simple_eng_c:
```

File: samples/buck_output_advanced/sample.yaml [hunk 17]
```diff
@@ -0,0 +1,31 @@
...
+common:
+  integration_platforms:
+    - nrf5340dk_nrf5340_cpuapp
+    - nrf52840dk_nrf52840
+  platform_allow: nrf5340dk_nrf5340_cpuapp nrf52840dk_nrf52840 nrf52dk_nrf52832
+  tags: pmic
+  timeout: 30
+
+tests:
+  sample.buck.output_voltage_eng_c:
```
Comment (wilu-nordic) [inline samples/buck_output_advanced/sample.yaml:20]:
```suggestion
  sample.buck_output_advanced_eng_c:
```

File: samples/buck_pins/sample.yaml [hunk 18]
```diff
@@ -0,0 +1,30 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+sample:
+  description: Sample for nPM1300 PMIC BUCKs
```
Comment (wilu-nordic) [inline samples/buck_pins/sample.yaml:8]:
```suggestion
  description: Buck pins sample
```

File: samples/buck_pins/sample.yaml [hunk 19]
```diff
@@ -0,0 +1,30 @@
...
+common:
+  integration_platforms:
+    - nrf5340dk_nrf5340_cpuapp
+    - nrf52840dk_nrf52840
+  platform_allow: nrf5340dk_nrf5340_cpuapp nrf52840dk_nrf52840 nrf52dk_nrf52832
+  tags: pmic
+  timeout: 30
+
+tests:
+  sample.buck.enable_bucks_using_pin_eng_c:
```
Comment (wilu-nordic) [inline samples/buck_pins/sample.yaml:20]:
```suggestion
  sample.buck_pins_eng_c:
```

File: samples/buck_pins/src/main.c [hunk 20]
```diff
@@ -0,0 +1,122 @@
...
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+#include <npmx_buck.h>
+#include <npmx_driver.h>
+#include <npmx_gpio.h>
+
+#define LOG_MODULE_NAME pmic_buck_pin
```
Comment (wilu-nordic) [inline samples/buck_pins/src/main.c:13]:
```suggestion
#define LOG_MODULE_NAME buck_pins
```

File: samples/buck_retention/sample.yaml [hunk 21]
```diff
@@ -0,0 +1,30 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+sample:
+  description: Sample for nPM1300 PMIC BUCKs
```
Comment (wilu-nordic) [inline samples/buck_retention/sample.yaml:8]:
```suggestion
  description: Buck retention sample
```

File: samples/buck_retention/src/main.c [hunk 22]
```diff
@@ -0,0 +1,101 @@
...
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+#include <npmx_buck.h>
+#include <npmx_driver.h>
+#include <npmx_gpio.h>
+
+#define LOG_MODULE_NAME pmic_buck_ret
```
Comment (wilu-nordic) [inline samples/buck_retention/src/main.c:13]:
```suggestion
#define LOG_MODULE_NAME buck_retention
```

File: samples/buck_output_simple/src/main.c [hunk 23]
```diff
@@ -0,0 +1,93 @@
...
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+#include <npmx_buck.h>
+#include <npmx_driver.h>
+
+#define LOG_MODULE_NAME pmic_buck_simple
```
Comment (wilu-nordic) [inline samples/buck_output_simple/src/main.c:12]:
```suggestion
#define LOG_MODULE_NAME buck_output_simple
```

**Comment by nordic-segl** [comment]
Waiting for results from integration CI
https://jenkins-ncs.nordicsemi.no/job/HCS/job/NPMX/job/npmx_zephyr_integration/job/main/439/

**Review by nordic-segl — APPROVED** [review]

File: samples/buck_pins/boards/nrf5340dk_nrf5340_cpuapp.overlay (hunk 11 above)
Comment (magp-nordic) [reply samples/buck_pins/boards/nrf5340dk_nrf5340_cpuapp.overlay:2]: Bump

**Comment by magp-nordic** [comment]
@jaz1-nordic It looks like some corrections were undone after a rebase. At least @wilu-nordic comments are applicable again.

**Review by nordic-segl — DISMISSED** [review]

File: samples/buck_output_advanced/README.rst [hunk 24]
```diff
@@ -0,0 +1,100 @@
...
+
+Overview
+********
+
+This sample uses the npmx BUCK driver to set advanced output voltage for the buck converter.
+The BUCK1 starts from voltage 1.0 V (:c:enumerator:`NPMX_BUCK_VOLTAGE_1V0`).
+The output voltage increases from the selected voltage to the maximum one (3.3 V).
+
+Voltage can be changed only for BUCK1.
+BUCK2 voltage is set by default to 3.0 V.
```
Comment (annakielar) [inline samples/buck_output_advanced/README.rst:29]:
```suggestion
This sample uses the npmx BUCK driver to set advanced output voltage for the buck converter.'

The output voltage for BUCK1 starts from 1.0 V (:c:enumerator:`NPMX_BUCK_VOLTAGE_1V0`) and increases by the selected value to the maximum 3.3 V.
Voltage can be changed only for BUCK1.
BUCK2 voltage is set by default to 3.0 V.
```
Please check if correct.

File: samples/buck_output_advanced/README.rst [hunk 25]
```diff
@@ -0,0 +1,100 @@
...
+
+Wiring
+******
+
+#. Connect the TWI interface between the chosen DK and the nPM1300 EK as in the following table:
```
Comment (annakielar) [inline samples/buck_output_advanced/README.rst:34]:
```suggestion
1. Connect the TWI interface between the chosen DK and the nPM1300 EK as in the following table:
```
Both versions are technically correct but we decided to use this variant for numbering.

File: samples/buck_output_advanced/README.rst [hunk 26]
```diff
@@ -0,0 +1,100 @@
...
+#. Connect a multimeter to VOUT1 that is the output voltage pin of BUCK1.
+#. |connect_kit|
+#. |connect_terminal|
+#. Check output voltage on EK **VOUT1** pin - it should start from 1.0 V and increase by 200 ms up to 3.3 V.
+#. Verify output voltage on BUCK1.
+
+Dependencies
+************
+
+This sample uses drivers from npmx.
```
Comment (annakielar) [inline samples/buck_output_advanced/README.rst:100]:
```suggestion
This sample uses drivers from npmx.
```
Would be great if we can be more specific (discussed offline).

File: samples/buck_output_simple/README.rst [hunk 27]
```diff
@@ -0,0 +1,99 @@
+.. _buck_output_simple_sample:
+
+BUCK - set simple output voltage
+################################
```
Comment (annakielar) [inline samples/buck_output_simple/README.rst:4]:
```suggestion
BUCK - Simple output voltage setting
####################################
```

File: samples/buck_output_simple/README.rst [hunk 28]
```diff
@@ -0,0 +1,99 @@
...
+
+.. table-from-sample-yaml::
+
+The sample also requires an nPM1300 EK.
+
+Overview
+********
+
+This sample uses the npmx BUCK driver to set the output voltage for the buck converter.
+The voltage is set to 2.4 V (:c:enumerator:`NPMX_BUCK_VOLTAGE_2V4`) on BUCK1.
```
Comment (annakielar) [inline samples/buck_output_simple/README.rst:25]:
```suggestion
You can set the output voltage to 2.4 V (:c:enumerator:`NPMX_BUCK_VOLTAGE_2V4`) on BUCK1.
```
Please verify.

File: samples/buck_pins/README.rst [hunk 29]
```diff
@@ -0,0 +1,104 @@
+.. _buck_pins_sample:
+
+BUCK - enable bucks using pin
+#############################
```
Comment (annakielar) [inline samples/buck_pins/README.rst:4]:
```suggestion
BUCK - enabling bucks using pin
###############################
```

File: samples/buck_output_advanced/README.rst [hunk 30]
```diff
@@ -0,0 +1,100 @@
+.. _buck_output_advanced_sample:
+
+BUCK - set advanced output voltage
+##################################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The BUCK Output Advanced sample demonstrates how to control PMIC BUCKs using npmx drivers.
```
[the remark this answers is not in the document]
Comment (annakielar) [reply samples/buck_output_advanced/README.rst:10]: Please use the same sentence structure for all 4 samples.

File: samples/buck_retention/README.rst [hunk 31]
```diff
@@ -0,0 +1,104 @@
+.. _buck_retention_sample:
+
+BUCK - set retention voltage
+############################
```
Comment (annakielar) [inline samples/buck_retention/README.rst:4]:
```suggestion
BUCK - Retention voltage setting
################################
```

**Review by annakielar — CHANGES_REQUESTED** [review]

**Review by magp-nordic — APPROVED** [review]

File: samples/buck_output_advanced/README.rst [hunk 32]
```diff
@@ -0,0 +1,100 @@
...
+
+Testing
+=======
+
+|test_sample|
+
+#. Connect a multimeter to VOUT1 that is the output voltage pin of BUCK1.
+#. |connect_kit|
+#. |connect_terminal|
+#. Check output voltage on EK **VOUT1** pin - it should start from 1.0 V and increase by 200 ms up to 3.3 V.
```
Comment (annakielar) [inline samples/buck_output_advanced/README.rst:94]:
```suggestion
#. Check output voltage on EK **VOUT1** pin - it should start from 1.0 V and increase every 200 ms up to 3.3 V.
```

File: samples/buck_output_advanced/README.rst [hunk 33]
```diff
@@ -0,0 +1,100 @@
...
+.. table-from-sample-yaml::
+
+The sample also requires an nPM1300 EK.
+
+Overview
+********
+
+This sample uses the npmx BUCK driver to set advanced output voltage for the buck converter.
+The BUCK1 starts from voltage 1.0 V (:c:enumerator:`NPMX_BUCK_VOLTAGE_1V0`).
+The output voltage increases from the selected voltage to the maximum one (3.3 V).
```
Comment (annakielar) [inline samples/buck_output_advanced/README.rst:26]:
```suggestion
The BUCK1 starts from 1.0 V (:c:enumerator:`NPMX_BUCK_VOLTAGE_1V0`), and the output voltage increases gradually every 200 ms to 3.3 V (:c:enumerator:`NPMX_BUCK_VOLTAGE_MAX`).
```

**Review by annakielar — CHANGES_REQUESTED** [review]

**Review by wilu-nordic — APPROVED** [review]

File: samples/buck_pins/README.rst [hunk 34]
```diff
@@ -0,0 +1,95 @@
+.. _buck_pins_sample:
+
+BUCK - Enabling bucks using pin
+###############################
+
+.. contents::
+   :local:
+   :depth: 2
+
+This BUCK sample variant demonstrates how to control PMIC BUCKs with the use of npmx drivers.
```
Comment (annakielar) [inline samples/buck_pins/README.rst:10]:
```suggestion
This BUCK sample variant demonstrates how to control the enable mode for the buck converter with GPIO with the use of npmx drivers.
```

File: samples/buck_output_advanced/README.rst [hunk 35]
```diff
@@ -0,0 +1,90 @@
...
+
+The sample also requires an nPM1300 EK.
+
+Overview
+********
+
+This sample uses the npmx BUCK driver to set advanced output voltage for the buck converter.
+
+The BUCK1 starts from 1.0 V (:c:enumerator:`NPMX_BUCK_VOLTAGE_1V0`), and the output voltage increases gradually every 200 ms to 3.3 V (:c:enumerator:`NPMX_BUCK_VOLTAGE_MAX`).
+Voltage can be changed only for BUCK1.
```
Comment (annakielar) [inline samples/buck_output_advanced/README.rst:27]:
```suggestion
The BUCK1 starts from 1.0 V (:c:enumerator:`NPMX_BUCK_VOLTAGE_1V0`), and the output voltage increases gradually every 200 ms to 3.3 V (:c:enumerator:`NPMX_BUCK_VOLTAGE_MAX`).

Voltage can be changed only for BUCK1.
```

**Review by annakielar — CHANGES_REQUESTED** [review]

**Review by magp-nordic — APPROVED** [review]

**Review by annakielar — APPROVED** [review]

**Review by wilu-nordic — APPROVED** [review]

**Review by masz-nordic — APPROVED** [review]

**Pull request merged**